k-distribution/src/main/scripts/lib/pyk/pyk/kastManip.py

```diff
@@ -208,32 +161,63 @@ def addOccurance(k):
         if match(pattern, k):
             occurances.append(k)
 
-    collectBottomUp(kast, addOccurance)
+    collect(addOccurance, kast)
     return occurances
 
 
-def countVarOccurances(kast, numOccurances=None):
-    """Count the number of occurances of each variable in a proof.
+def extract_lhs(term: KInner) -> KInner:
+    return top_down(if_ktype(KRewrite, lambda rw: rw.lhs), term)
+
+
+def extract_rhs(term: KInner) -> KInner:
+    return top_down(if_ktype(KRewrite, lambda rw: rw.rhs), term)
 
-    -   Input: Kast term.
-    -   Output: Map of variable names to their number of occurances.
-    """
-    numOccurances = {} if numOccurances is None else numOccurances
 
-    def _getNumOccurances(_kast):
-        if type(_kast) is KVariable:
-            vName = _kast.name
-            if vName in numOccurances:
-                numOccurances[vName] += 1
-            else:
-                numOccurances[vName] = 1
+def count_vars(term: KInner) -> Counter:
+    counter: Counter = Counter()
 
-    collectBottomUp(kast, _getNumOccurances)
-    return numOccurances
+    def count(term: KInner) -> None:
+        if type(term) is KVariable:
+            counter[term.name] += 1
+
+    collect(count, term)
+    return counter
+
+
+def count_rhs_vars(term: KInner) -> Counter:
+    def recur(term: KInner, *, rhs=False) -> Counter:
```

Comment:
What does the second argument `*,` mean here?

Reply:
Arguments that follow `*` are [keyword-only arguments](https://peps.python.org/pep-3102/). In this case it means for example that `recur(term, True)` is not syntactically correct, only `recur(term, rhs=True)` is.